main/appserver/admingui/core/src/main/java/org/glassfish/admingui/handlers/WoodstockHandler.java

```diff
@@ -104,12 +104,13 @@ public WoodstockHandler() {
                 @HandlerInput(name = "deleteTempFile", type = String.class)})
     public static void deleteFileFromTempDir(HandlerContext handlerCtx) throws IOException {
         Logger logger = GuiUtil.getLogger();
+        String gfHome = System.getProperty("com.sun.aas.installRoot");
         if (logger.isLoggable(Level.FINE)) {
             logger.fine(GuiUtil.getCommonMessage("log.indeleteFileFromTempDir"));
         }
 
         String deleteTempFile = (String) handlerCtx.getInputValue("deleteTempFile");
-        if (deleteTempFile == null) {
+        if (deleteTempFile == null  || !deleteTempFile.contains(gfHome)) {
```

Comment:
Checking for installRoot is not right since the actual temporary directory is configurable. Consider the case where a Java EE sample (found under <as-install>/samples) is deployed as a local package, the app file will still get deleted.